### PR TITLE
Separate IO from txBuild and txBuildRaw functions

### PR DESCRIPTION
**Pull request by Jimbo4350**
First step towards condensing modules responsible for reading and validating the things required for transaction construction. The end goal is to expose these modules via `cardano-api` so users can implement their own transaction construction and/or validation functions.

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Fees.hs [hunk 1]
```diff
@@ -1025,17 +1026,18 @@ makeTransactionBodyAutoBalance eraInMode systemstart history pparams
...
           txFee  = TxFeeExplicit explicitTxFees fee,
           txOuts = accountForNoChange
                      (TxOut changeaddr balance TxOutDatumNone ReferenceScriptNone)
                      (txOuts txbodycontent),
           txReturnCollateral = retColl,
           txTotalCollateral = reqCol
         }
-    return (BalancedTxBody txbody3 (TxOut changeaddr balance TxOutDatumNone ReferenceScriptNone) fee)
+    txbody3 <-
+      first TxBodyError $ -- TODO: impossible to fail now
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/Fees.hs:1038]: I think the TODO comment needs elaboration.  The TODO should say what needs to be done.

Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/Fees.hs:1038]: That was there previously but I'll update it

**Review by newhoggy — APPROVED** [review]

File: cardano-api/src/Cardano/Api/Fees.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-api/src/Cardano/Api/Fees.hs]: Thanks!

**Pull request merged**